src/centralserver/admin-service/src/main/java/org/niis/xroad/centralserver/restapi/service/SystemParameterService.java

```diff
@@ -0,0 +1,133 @@
+    public static final String SECURITY_SERVER_OWNERS_GROUP = "securityServerOwnersGroup";
+    public static final String DEFAULT_SECURITY_SERVER_OWNERS_GROUP = "security-server-owners";
+    public static final String DEFAULT_SECURITY_SERVER_OWNERS_GROUP_DESC = "Security server owners";
+    public static final String CONF_SIGN_DIGEST_ALGO_ID = "confSignDigestAlgoId";
+    public static final String DEFAULT_CONF_SIGN_DIGEST_ALGO_ID = CryptoUtils.SHA512_ID;
+    public static final String OCSP_FRESHNESS_SECONDS = "ocspFreshnessSeconds";
+    public static final Integer DEFAULT_OCSP_FRESHNESS_SECONDS = 3600;
+    public static final String TIME_STAMPING_INTERVAL_SECONDS = "timeStampingIntervalSeconds";
+    public static final Integer DEFAULT_TIME_STAMPING_INTERVAL_SECONDS = 60;
+    public static final String CONF_EXPIRE_INTERVAL_SECONDS = "confExpireIntervalSeconds";
+    public static final Integer DEFAULT_CONF_EXPIRE_INTERVAL_SECONDS = 600;
+    private static final String[] NODE_LOCAL_PARAMETERS = {CENTRAL_SERVER_ADDRESS};
+
+    private final SystemParameterRepository systemParameterRepository;
+    private final HAConfigStatus currentHaConfigStatus;
+
+    public String getParameterValue(String key, String defaultValue) {
+        Optional<SystemParameter> valueInDb;
+        log.debug("getParameterValue() - getting value for key:{} with default value:{}", key, defaultValue);
+        if (currentHaConfigStatus.isHaConfigured()
+                && isNodeLocalParameter(key)
+        ) {
+            valueInDb = systemParameterRepository.findByKeyAndHaNodeName(
+                    key,
+                    currentHaConfigStatus.getCurrentHaNodeName()
+            );
+        } else {
+            List<SystemParameter> systemParameters = systemParameterRepository.findByKey(key);
+            valueInDb = systemParameters.stream().findFirst();
+            assert 1 >= systemParameters.size();
+        }
+        if (valueInDb.isPresent()) {
+            String valueFromDb = valueInDb.get().getValue();
+            log.trace("getParameterValue found a {} SystemParameter value {}  ", key, valueFromDb);
+            return valueFromDb;
+        }
+        log.trace("getParameterValue returns given default value for key {}", key);
+        return defaultValue;
+    }
+
+    public SystemParameter updateOrCreateParameter(String lookupKey, String updateValue) {
+        Optional<SystemParameter> systemParameter;
+        if (currentHaConfigStatus.isHaConfigured() && isNodeLocalParameter(lookupKey)) {
+            String haNodeName = currentHaConfigStatus.getCurrentHaNodeName();
+            systemParameter = systemParameterRepository.findByKeyAndHaNodeName(lookupKey, haNodeName)
+                    .stream().findFirst();
+
+        } else {
+            systemParameter = systemParameterRepository.findByKey(lookupKey)
+                    .stream().findFirst();
+        }
```

Comment:
Same logic in "getParameterValue", could be refactored to helper function.

Reply:
c0772c7